### PR TITLE
[vmss create] remove unnecessary client side validation

### LINKED ISSUE #9630: Unable to create a VMSS in single-placement-group larger than 100 nodes
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
I am trying to create a VMSS larger than 100 instances with RDMA (all H series node types).  This is possible using the `az vmss scale` but a hard-coded limit stops this with `az vmss create`.

**To Reproduce**
Create a VMSS with `--single-placement-group true` and `--instance-count` greater than 100. 
 The error is:

    ERROR: usage error: '--single-placement-group' should be turned off for zonal scale-sets or with 100+ instances

We require a single placement group for the H series node types to all be on the same IB network.

**Expected behavior**
The VMSS should be created (the limit is now 300 AFAIK).

**Environment summary**
Just az cli.

**Additional context**
Here is the hardcoded limit in the code:
    https://github.com/Azure/azure-cli/blob/3bdff7e063121af9c5528c2689413cbf6d213886/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L1096

### PR DESCRIPTION
**Pull request by adewaleo**
closes #9630

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by marstr** [comment]
I believe you fell victim to https://github.com/Azure/azure-cli/issues/9648. Please rebase on top of [`dev`](https://github.com/Azure/azure-cli/tree/dev) (d2cf98f) to adopt the newest changes to CI, which should unblock your contribution.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/hybrid_2018_03_01/test_vm_defaults.py [hunk 1]
```diff
@@ -340,50 +340,6 @@ class TestBigVMSSDefaults(unittest.TestCase):
     def _set_up_ns(cls, ns):
         ns.single_placement_group, ns.zones, ns.platform_fault_domain_count, ns.instance_count = None, None, None, 2
 
-    def test_vmss_disable_single_placement_group(self):
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/hybrid_2018_03_01/test_vm_defaults.py]: Why are we removing these tests?  Single placement group is still a feature, so we should still keep the tests in place. The only thing we would not need to test is the validation you removed.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -327,7 +327,7 @@ def load_arguments(self, _):
         VirtualMachineEvictionPolicyTypes = self.get_models('VirtualMachineEvictionPolicyTypes', resource_type=ResourceType.MGMT_COMPUTE)
         c.argument('name', name_arg_type)
         c.argument('nat_backend_port', default=None, help='Backend port to open with NAT rules.  Defaults to 22 on Linux and 3389 on Windows.')
-        c.argument('single_placement_group', arg_type=get_three_state_flag(), help="Enable replicate using fault domains within the same cluster. Default to 'false' for any zonals, or with 100+ instances"
+        c.argument('single_placement_group', arg_type=get_three_state_flag(), help="Limit the scale set to a single placement group with a max size of 100 instances."
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Is the 100 instance limit still applicable?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: I don't know the answer, but per conversation with CRP, we shall remove all the client side validations on this, and leverage the server end error. So @adewaleo, please try out and see whether or not this limitation is no longer valid, or the server error is clear now

Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: I will try out. I was just wary of creating a VMSS with a lot of instances. 

Forgot to update the help text. Will test and also update the help text.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/hybrid_2018_03_01/test_vm_defaults.py (hunk 1 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/tests/hybrid_2018_03_01/test_vm_defaults.py]: These tests were testing the client side validation I removed. 

I could try to add tests for the parts remaining in the validator

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py (hunk 2 above)
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: cc: @yugangw-msft

**Comment by yugangw-msft** [comment]
Please remove the ` src/command_modules/azure-cli-vm/HISTORY.rst` and merge

**Comment by adewaleo** [comment]
I need an approval @limingu @tjprescott ?

**Pull request merged**